serverless-code-commit-repo-event-notifier.py

```diff
@@ -52,52 +52,55 @@ def post_to_slack(webhook_url, slack_data):
         tmp = {}
-        tmp["fallback"]         = "Event Detected."
-        tmp["color"]            = i.get("color")
-        tmp["pretext"]          = f"CodeCommit Event detected in `{i.get('awsRegion')}` in Repo:`{i.get('eventSourceARN').split(':')[-1]}`"
-        tmp["author_name"]      = "Serverless-Repo-Monitor"
-        tmp["author_link"]      = "https://github.com/miztiik/serverless-code-commit-repo-event-notifier"
-        tmp["author_icon"]      = "https://avatars1.githubusercontent.com/u/12252564?s=400&u=20375d438d970cb22cc4deda79c1f35c3099f760&v=4"
-        tmp["title"]            = f"Repo Event: {i.get('eventName')}"
-        tmp["title_link"]       = f"https://console.aws.amazon.com/codesuite/codecommit/repositories/{i.get('eventSourceARN').split(':')[-1]}/browse?region={i.get('awsRegion')}"
-        tmp["fields"]           = [
-                    {
-                        "title": "Repo Name",
-                        "value": f"`{i.get('repo_metadata').get('repositoryName')}`",
-                        "short": True
-                    },
-                        "title": "Repo Default Branch",
-                        "value": f"`{i.get('repo_metadata').get('defaultBranch')}`",
-                        "short": True
-                    },
-                    {
-                        "title": "Triggered-By",
-                        "value": f"`{i.get('userIdentityARN')}`",
-                        "short": True
-                    },
-                    {
-                        "title": "Trigger Name",
-                        "value": f"`{i.get('eventTriggerName')}`",
-                        "short": True
-                    }
-                ]
-        tmp["footer"]           = "AWS CodeCommit"
-        tmp["footer_icon"]      = "https://raw.githubusercontent.com/miztiik/serverless-code-commit-repo-event-notifier/master/images/aws-code-commit-logo.png"
-        tmp["ts"]               = int( dateutil.parser.parse( i.get('eventTime') ).timestamp()  )
-        tmp["mrkdwn_in"]        = ["pretext", "text", "fields"]
+        tmp["fallback"] = "Event Detected."
+        tmp["color"] = i.get("color")
+        tmp["pretext"] = f"CodeCommit Event detected in `{i.get('awsRegion')}` in Repo:`{i.get('eventSourceARN').split(':')[-1]}`"
+        tmp["author_name"] = "Serverless-Repo-Monitor"
+        tmp["author_link"] = "https://github.com/miztiik/serverless-code-commit-repo-event-notifier"
+        tmp["author_icon"] = "https://avatars1.githubusercontent.com/u/12252564?s=400&u=20375d438d970cb22cc4deda79c1f35c3099f760&v=4"
+        tmp["title"] = f"Repo Event: {i.get('eventName')}"
+        tmp["title_link"] = f"https://console.aws.amazon.com/codesuite/codecommit/repositories/{i.get('eventSourceARN').split(':')[-1]}/browse?region={i.get('awsRegion')}"
+        tmp["fields"] = [
+            {
+                "title": "Repo Name",
+                "value": f"`{slack_data.get('repo_metadata').get('repositoryName')}`",
```

Comment:
I haven't run it again, But
You are accessing the list element `slack.data` directly inside the for loop. Maybe the list has only one item and it works fine now, but if the list has more than one item, this will fail.

in `tmp["color"] = i.get("color")` you are still accessing the iterated value. You are mixing both. Can you either remove the for loop or find out why the for loop isnt producing the actual results?

Reply:
I can get the 'slack.data's out of the loop, but I can't agree with the idea that it will be failure if the last has more than one item. Yes, It is obviously redundant, but it won't fail. 
More over, you don't need to worry about "more than one item" in my opinion. I've never seen "more than one item". If there's more than one item with information of more than one repository, this logic would be changed a lot. 

The json blow is captured by executing with a real codecommit event. 
`
{
  "Records": [
    {
      "awsRegion": "ap-northeast-2",
      "codecommit": {
        "references": [
          {
            "commit": "72217cfe5cf20bf7e8e7a0998200ab23092d04fd",
            "ref": "refs/heads/master"
          }
        ]
      },
      "customData": "",
      "eventId": "ff3ef7d3-4a1b-40a8-94e0-2ef2333a2ba4",
      "eventName": "ReferenceChanges",
      "eventPartNumber": 1,
      "eventSource": "aws:codecommit",
      "eventSourceARN": "arn:aws:codecommit:ap-northeast-2:932892212733:test",
      "eventTime": "2020-02-17T00:55:09.490+0000",
      "eventTotalParts": 1,
      "eventTriggerConfigId": "5b6285a9-eacf-428e-89e7-da114ea0cf0d",
      "eventTriggerName": "test",
      "eventVersion": "1.0",
      "userIdentityARN": "arn:aws:iam::932892212733:user/hoon.choi@nxc.com"
    }
  ],
  "repo_metadata": {
    "accountId": "932892212733",
    "repositoryId": "9057bc9e-e837-43f5-9367-57bd6bf957c1",
    "repositoryName": "test",
    "defaultBranch": "master",
    "lastModifiedDate": "datetime.datetime(2020, 2, 17, 0, 55, 9, 416000, tzinfo=tzlocal())",
    "creationDate": "datetime.datetime(2020, 2, 17, 0, 38, 57, 400000, tzinfo=tzlocal())",
    "cloneUrlHttp": "https://git-codecommit.ap-northeast-2.amazonaws.com/v1/repos/test",
    "cloneUrlSsh": "ssh://git-codecommit.ap-northeast-2.amazonaws.com/v1/repos/test",
    "Arn": "arn:aws:codecommit:ap-northeast-2:932892212733:test"
  }
}
`

Reply:
> you don't need to worry about "more than one item" in my opinion

What if there are multiple devs committing to the same repo at the same time. Is the event triggers from code commit to lambda, 1-to-1 mapping,
or 
does it have a digest of all the commits from a merge/PR. 
Need to check the documentation on this and a better test case to see if ever get an event with more than one item. After that we can remove the `for` loop completely.

Reply:
I can get your concern I will make additional pull request :) 
one record-to-one slack message

Reply:
@miztiik please check that I committed